scio-extra/src/main/scala/com/spotify/scio/extra/csv/package.scala

```diff
@@ -18,5 +18,38 @@
 package com.spotify.scio.extra
 
 import com.spotify.scio.extra.csv.syntax.AllSyntax
+import kantan.csv.ops.toCsvOutputOps
+import kantan.csv.{CsvConfiguration, CsvWriter, HeaderEncoder}
+import org.apache.beam.sdk.io.FileIO
 
-package object csv extends AllSyntax
+import java.io.Writer
+import java.nio.channels.{Channels, WritableByteChannel}
+import java.nio.charset.StandardCharsets
+
+/**
+ * Main package for CSV type-safe APIs. Import all.
+ *
+ * {{{
+ * import com.spotify.scio.extra.csv._
+ * }}}
+ */
+package object csv extends AllSyntax {
+  final private[scio] class CsvSink[T: HeaderEncoder](csvConfig: CsvConfiguration)
```

Comment:
Defining classes in package object has been discouraged. This creates the following warning:
>it is not recommended to define classes/objects inside of package objects.
If possible, define class CsvSink in package csv instead.

It would be nicer to move this to another compile unit inside the csv package.

Reply:
Got it. Is there a plan to migrate other classes that are defined in package objects? e.g. https://github.com/klDen/scio/blob/d97f0387393315d89fd0bbd6684306bd5f37a376/scio-parquet/src/main/scala/com/spotify/scio/parquet/package.scala

Reply:
Yes, this is a work in progress: https://github.com/spotify/scio/pull/5127